plenum/test/node_catchup/test_process_catchup_replies.py

```diff
@@ -0,0 +1,156 @@
+from plenum.common.constants import DOMAIN_LEDGER_ID, LedgerState
+from plenum.common.messages.node_messages import CatchupRep
+from plenum.common.txn_util import append_txn_metadata, reqToTxn
+from plenum.common.types import f
+from plenum.common.util import SortedDict
+from plenum.test.helper import sdk_signed_random_requests
+
+ledger_id = DOMAIN_LEDGER_ID
+
+
+def _add_txns_to_ledger(node, looper, sdk_wallet_client, num_txns_in_reply, reply_count):
+    '''
+    Add txn_count transactions to node's ledger and return
+    ConsistencyProof for all new transactions and list of CatchupReplies
+    :return: ConsistencyProof, list of CatchupReplies
+    '''
+    txn_count = num_txns_in_reply * reply_count
+    ledger_manager = node.ledgerManager
+    ledger = ledger_manager.ledgerRegistry[ledger_id].ledger
+    ledger_info = ledger_manager.getLedgerInfoByType(ledger_id)
+    reqs = sdk_signed_random_requests(looper, sdk_wallet_client, txn_count)
+    # add transactions to ledger
+    for req in reqs:
+        txn = append_txn_metadata(reqToTxn(req), txn_time=12345678)
+        ledger_manager._add_txn(
+            ledger_id, ledger, ledger_info, txn)
+    # generate CatchupReps
+    replies = []
+    for i in range(reply_count, 0, -1):
```

Comment:
Why the order is still reverse?

Reply:
If it will be not reversed, in the for we will have:
`start = ledger.seqNo + 1 - (txn_count -  i) * num_txns_in_reply`
`end = ledger.seqNo - (txn_count - i - 1) * num_txns_in_reply`
I think, it looks worse.

Reply:
Maybe it would be better to just iterate over `range(ledger.seqNo - txn_count + 1, ledger.seqNo + 1, num_txns_in_reply)`?

Reply:
I'm not sure that it is more understandable. But thank you, changed it.